akka-http-core/src/main/scala/akka/http/javadsl/settings/ServerSentEventSettings.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.javadsl.settings
+
+import akka.annotation.{ ApiMayChange, DoNotInherit }
+import akka.http.impl.settings.ServerSentEventSettingsImpl
+
+/**
+ * Public API but not intended for subclassing
+ *
+ * Options that are in "preview" or "early access" mode.
+ * These options may change and/or be removed within patch releases
+ * without early notice (e.g. by moving them into a stable supported place).
+ */
+@ApiMayChange @DoNotInherit
+abstract class ServerSentEventSettings private[akka] () { self: ServerSentEventSettingsImpl ⇒
+
+  /**
+   * The maximum size for parsing server-sent events
+   */
+  def maxEventSize: Int
+
+  /**
+   * The maximum size for parsing lines of a server-sent event
+   */
+  def maxLineSize: Int
+
+  // ---
+
+  def withMaxEventSize(newValue: Int): ServerSentEventSettingsImpl = self.copy(maxEventSize = newValue)
```

Comment:
The return type should be the public `ServerSentEventSettings` here.

```suggestion
  def withMaxEventSize(newValue: Int): ServerSentEventSettings = self.copy(maxEventSize = newValue)
```